### PR TITLE
Add `VAEBackbone` and use it for SD3

### PR DESCRIPTION
**Pull request by james77777778**
This PR adds the VAE encoder to SD3 and also fixes the summary of `Task` to recursively parse the info, rather than relying on the layer attribute.

The preset on kaggle has also been updated.

Demo colab:
https://colab.research.google.com/drive/1ucx2lEck1ZO3HVTjZ4qgVsQrBWrHWfVQ?usp=sharing

### CODE REVIEW

**Review by divyashreepathihalli — COMMENTED** [review]
Thanks @james77777778, left a few comments.

File: keras_hub/src/models/stable_diffusion_3/autoencoder.py [hunk 1]
```diff
@@ -0,0 +1,787 @@
+import math
+
+import keras
+from keras import ops
+
+from keras_hub.src.models.backbone import Backbone
+from keras_hub.src.utils.keras_utils import standardize_data_format
+
+
+class Attention(keras.layers.Layer):
```
[the remark this answers is not in the document]
Comment (james77777778) [reply keras_hub/src/models/stable_diffusion_3/autoencoder.py]: I have added the docstrings. This layer is similar to `MultiHeadAttention` but it uses `Conv2D` and `GroupNormalization`.

File: keras_hub/src/models/stable_diffusion_3/autoencoder.py [hunk 2]
```diff
@@ -0,0 +1,787 @@
...
+                "groups": self.groups,
+            }
+        )
+        return config
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+
+class ResNetBlock(keras.layers.Layer):
```
[the remark this answers is not in the document]
Comment (james77777778) [reply keras_hub/src/models/stable_diffusion_3/autoencoder.py]: The normalization layer and activation function are different.

File: keras_hub/src/models/stable_diffusion_3/autoencoder.py [hunk 3]
```diff
@@ -0,0 +1,787 @@
...
+        )
+        return config
+
+    def compute_output_shape(self, input_shape):
+        output_shape = list(input_shape)
+        output_shape[self.axis] = output_shape[self.axis] // 2
+        return output_shape
+
+
+class AutoEncoder(Backbone):
```
[the remark this answers is not in the document]
Comment (james77777778) [reply keras_hub/src/models/stable_diffusion_3/autoencoder.py]: Got it. This is better!
I have added a folder called `vae` and I believe it can be also used for FLUX models.

**Comment by james77777778** [comment]
@divyashreepathihalli 
This PR is ready for review. Please let me know if any further changes are required.

- Colab for demo: https://colab.research.google.com/drive/1ucx2lEck1ZO3HVTjZ4qgVsQrBWrHWfVQ

File: keras_hub/src/models/stable_diffusion_3/flow_match_euler_discrete_scheduler.py [hunk 4]
```diff
@@ -27,7 +27,7 @@ class FlowMatchEulerDiscreteScheduler(layers.Layer):
     https://arxiv.org/abs/2403.03206).
     """
 
-    def __init__(self, num_train_timesteps=1000, shift=1.0, **kwargs):
+    def __init__(self, num_train_timesteps=1000, shift=3.0, **kwargs):
```
Comment (james77777778) [inline keras_hub/src/models/stable_diffusion_3/flow_match_euler_discrete_scheduler.py:30]: I have found that we should use `3.0` here. The image quality is slightly improved and this value aligns with the settings in `huggingface/diffusers`.

File: keras_hub/src/models/stable_diffusion_3/stable_diffusion_3_backbone_test.py [hunk 5]
```diff
@@ -5,11 +5,22 @@
 from keras_hub.src.models.stable_diffusion_3.stable_diffusion_3_backbone import (
     StableDiffusion3Backbone,
 )
+from keras_hub.src.models.vae.vae_backbone import VAEBackbone
 from keras_hub.src.tests.test_case import TestCase
 
 
 class StableDiffusion3BackboneTest(TestCase):
     def setUp(self):
+        height, width = 64, 64
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/stable_diffusion_3/stable_diffusion_3_backbone_test.py:14]: since you have added the presets, you can add `self.run_presets_test`

**Review by divyashreepathihalli — APPROVED** [review]
LGTM

File: keras_hub/src/models/stable_diffusion_3/stable_diffusion_3_backbone_test.py (hunk 5 above)
Comment (james77777778) [reply keras_hub/src/models/stable_diffusion_3/stable_diffusion_3_backbone_test.py:14]: Added. The test passed on my setup.

File: keras_hub/src/models/task.py [hunk 6]
```diff
@@ -325,21 +328,20 @@ def add_layer(layer, info):
...
-            if image_converter:
-                info = "Image size: "
-                info += highlight_shape(image_converter.image_size())
-                add_layer(image_converter, info)
-            audio_converter = getattr(preprocessor, "audio_converter", None)
-            if audio_converter:
-                info = "Audio shape: "
-                info += highlight_shape(audio_converter.audio_shape())
-                add_layer(audio_converter, info)
+            if preprocessor and isinstance(preprocessor, keras.Layer):
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/task.py:331]: can you please explain why you have made these changes?

File: keras_hub/src/models/vae/vae_layers.py [hunk 7]
```diff
@@ -0,0 +1,740 @@
...
+            and outputs_shape[w_axis] is not None
+        ):
+            outputs_shape[h_axis] = outputs_shape[h_axis] // scale_factor
+            outputs_shape[w_axis] = outputs_shape[w_axis] // scale_factor
+        outputs_shape[c_axis] = self.output_channels
+        return outputs_shape
+
+
+class VAEDecoder(keras.layers.Layer):
+    """The decoder part of VAE.
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/vae/vae_layers.py:474]: NIT `part`->`layer`

**Review by divyashreepathihalli — COMMENTED** [review]
Look good! need some clarification on the changes to task.py. Otherwise LGTM!

File: keras_hub/src/models/task.py (hunk 6 above)
Comment (james77777778) [reply keras_hub/src/models/task.py:334]: This is needed for a nested preprocessor. I have added a comment for this change:

```python
            # Since the preprocessor might be nested with multiple `Tokenizer`,
            # `ImageConverter`, `AudioConverter` and even other `Preprocessor`
            # instances, we should recursively iterate through them.
```

File: keras_hub/src/models/vae/vae_layers.py (hunk 7 above)
Comment (james77777778) [reply keras_hub/src/models/vae/vae_layers.py]: Fixed.
